internal/k8s/validator.go

```diff
@@ -371,7 +372,25 @@ func (l *leader) ValidateAviInfraSetting(key string, infraSetting *akov1beta1.Av
 			refData[nodeNetwork.NetworkName] = "Network"
 		}
 	}
+	tenant := lib.GetTenant()
+	var err error
 	if infraSetting.Spec.SeGroup.Name != "" {
+		// In case of WCP/VCF deployments in VPC mode, NS can have SeGroup(shared/dedicated)
+		// annotation which results in setting SeGroup in infrasetting and for validation of SeGroup,
+		// tenant needs to be determined through vpc path in VPCCR.
+		if lib.GetVPCMode() && infraSetting.Spec.NSXSettings.T1LR != nil {
+			aviClientPool := cache.SharedAVIClients(lib.GetTenant())
+			if aviClientPool == nil || len(aviClientPool.AviClient) == 0 {
+				return fmt.Errorf("avi rest client initialization failed")
```

Comment:
This gets checked at the time of initial creation of aviClient for Admin tenant, we can skip this check here